lib/queueClient.ts

```diff
@@ -0,0 +1,209 @@
+ */
+export interface QueueClientOptions {
+  /**
+   * @property {number} [receiveMode] The mode in which messages should be received.
+   * Default: ReceiveMode.peekLock
+   */
+  receiveMode?: ReceiveMode;
+  /**
+   * @property {number} [maxConcurrentCalls] The maximum number of messages that should be
+   * processed concurrently while in peek lock mode. Once this limit has been reached, more
+   * messages will not be received until messages currently being processed have been settled.
+   * Default: 1
+   */
+  maxConcurrentCalls?: number;
+}
+
+export class QueueClient extends Client {
+  /**
+   * @property {number} receiveMode The mode in which messages should be received.
+   * Default: ReceiveMode.peekLock
+   */
+  receiveMode: ReceiveMode;
+  /**
+   * @property {number} maxConcurrentCalls The maximum number of messages that should be
+   * processed concurrently while in peek lock mode. Once this limit has been reached, more
+   * messages will not be received until messages currently being processed have been settled.
+   */
+  maxConcurrentCalls: number;
+
+  /**
+   * Instantiates a client pointing to the ServiceBus Queue given by this configuration.
+   *
+   * @constructor
+   * @param {string} name The Queue name.
+   * @param {ConnectionContext} context The connection context to create the QueueClient.
+   * @param {QueueClientOptions} [options] The QueueClient options.
+   */
+  constructor(name: string, context: ConnectionContext, options?: QueueClientOptions) {
+    super(name, context);
+    if (!options) options = {};
+    this.receiveMode = options.receiveMode || ReceiveMode.peekLock;
+    this.maxConcurrentCalls = options.maxConcurrentCalls != undefined ? options.maxConcurrentCalls : 1;
+  }
+
+  /**
+   * Closes the AMQP connection to the ServiceBus Queue for this client,
+   * returning a promise that will be resolved when disconnection is completed.
+   * @returns {Promise<any>}
+   */
+  async close(): Promise<any> {
+    try {
+      if (this._context.namespace.connection && this._context.namespace.connection.isOpen()) {
+        // Close the sender.
+        if (this._context.sender) {
+          await this._context.sender.close();
+        }
+        // Close the receiver.
+        if (this._context.streamingReceiver) {
+          await this._context.streamingReceiver.close();
+        }
+        // Close the management session
+        await this._context.managementSession!.close();
+        debug("Closed the client '%s'.", this.id);
+      }
+    } catch (err) {
+      const msg = `An error occurred while closing the queue client ` +
+        `"${this.id}": ${JSON.stringify(err)} `;
+      debug(msg);
+      throw new Error(msg);
+    }
+  }
+
+  /**
+   * Sends the given message to the ServiceBus Queue.
+   *
+   * @param {any} data  Message to send.  Will be sent as UTF8-encoded JSON string.
+   * @returns {Promise<Delivery>} Promise<Delivery>
+   */
+  async send(data: SBMessage): Promise<Delivery> {
```

Comment:
what's contained in a `Delivery`? it looks like the type comes all the way from the rhea library. i'm not sure what data is needed besides the error if something goes wrong. unless there's a strong need for it, i'd just return `Promise<void>` until you need something else

Reply:
People might need Delivery for tags associated with it. Hence keeping it over there.